### ISSUE TITLE
Migration with foreign key failing with two separate calls to runMigration

### ISSUE THREAD

**Issue by bergmark** [yesodweb/persistent#152]
Description copied from the [google group](https://groups.google.com/forum/#!topic/yesodweb/0EySKveHrZk)

I have a pretty simple test case with two models (in separate modules) with a foreign key constraint, https://gist.github.com/bergmark/6539373

I make a call to runMigration for each module and then get the error "Table not found: A" on the second migration. If i put both models in the same persist block this problem doesn't occur. This is with the latest hackage versions of persistent, persistent-template, persistent-sqlite (and persistent-postgresql).

**Comment by gregwebs** [comment]
I did cleanup some migrations (so try the latest version), but I have not tested cross-module behavior.

**Comment by snoyberg** [comment]
Closing out old issues, please reopen if still relevant

**Comment by parsonsmatt** [comment]
There's a work-around for this, documented in [my blog post](https://www.parsonsmatt.org/2019/12/06/splitting_persistent_models.html). 

I'd think the problem might be solved by running:

```haskell
runMigration $ do
  A.migrateAll
  B.migrateAll
```

instead of

```haskell
runMigration A.migrateAll
runMigration B.migrateAll
```

but I'll need to experiment with it.

**Comment by parsonsmatt** [comment]
Fixed by #1256
